### PR TITLE
Add read flag to alerts

### PR DESCRIPTION
**Pull request by ivnsch**

### CODE REVIEW

File: src/database/alert_dao.rs [hunk 1]
```diff
@@ -550,4 +593,163 @@ mod tests {
...
+        assert!(loaded_alerts_res.is_ok());
+
+        let loaded_alerts = loaded_alerts_res.unwrap();
+
+        assert_eq!(loaded_alerts.len(), 1);
+        assert_eq!(
+            loaded_alerts[0],
+            Alert {
+                is_read: true,
+                ..alert
```
Comment (duskoo) [inline src/database/alert_dao.rs:645]: [Struct update syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax), nice :)

File: src/database/alert_dao.rs [hunk 2]
```diff
@@ -550,4 +593,163 @@ mod tests {
...
+            contact_end: 2000,
+            min_distance: 2.3,
+            avg_distance: 4.3,
+            is_read: false,
+        };
+
+        let save_res = alert_dao.save(vec![alert.clone()]);
+        assert!(save_res.is_ok());
+
+        let delete_res = alert_dao.update_is_read("1".to_owned(), true);
```
Comment (duskoo) [inline src/database/alert_dao.rs]: update_res would be better name (there is not deletion taking place)

File: src/database/alert_dao.rs [hunk 3]
```diff
@@ -550,4 +593,163 @@ mod tests {
         assert_eq!(loaded_alerts.len(), 1);
         assert_eq!(loaded_alerts[0], alert1);
     }
+
+    #[test]
+    fn test_marks_alert_as_read() {
+        let database = Arc::new(Database::new(
+            Connection::open_in_memory().expect("Couldn't create database!"),
+        ));
+        let alert_dao = AlertDaoImpl::new(database.clone());
```
Comment (duskoo) [inline src/database/alert_dao.rs:602]: immutable api preferred, right?

**Review by duskoo — APPROVED** [review]

File: src/database/alert_dao.rs (hunk 3 above)
Comment (ivnsch) [reply src/database/alert_dao.rs:602]: In Rust not necessarily, as mutability is safer than Swift/Kotlin.

Not sure how it relates with this code though?

Comment (duskoo) [reply src/database/alert_dao.rs:602]: @i-schuetz 
> In Rust not necessarily, as mutability is safer than Swift/Kotlin.
> 
> Not sure how it relates with this code though?

I was wondering why you prefer passing _database.clone()_ as argument, as opposed to just _database_ . Is it due adopted project convention, or ...?

Comment (ivnsch) [reply src/database/alert_dao.rs:602]: Ah, no, in this case it's probably a copy-paste leftover from the database injection in the composition root. Cloning is not needed here as it's only used by `AlertDaoImpl`

Comment (ivnsch) [reply src/database/alert_dao.rs:602]: https://github.com/Co-Epi/app-backend-rust/pull/143

**Pull request merged**